client/client_test.go

```diff
@@ -1475,3 +1475,126 @@ func TestRemoteServerUnavailableNoLocalCache(t *testing.T) {
 	assert.Error(t, err)
 	assert.IsType(t, store.ErrServerUnavailable{}, err)
 }
+
+// AddDelegation creates a valid changefile (rejects invalid delegation names),
+// which when applied, adds a new delegation role with the correct keys.
+func TestAddDelegation(t *testing.T) {
```

Comment:
To verify that AddDelegation doesn't validate the hierarchy, you mean?  Sounds good, will change it.  :)